custom_components/magiqtouch/climate.py

```diff
@@ -83,24 +101,56 @@ async def async_setup_entry(
     """Set up BSBLan device based on a config entry."""
     driver: MagiQtouch_Driver = hass.data[DOMAIN][entry.entry_id]
     await driver.login()
-    await hass.async_add_executor_job(
-        driver.mqtt_connect,
-    )
-    async_add_entities([MagiQtouch(entry.entry_id, driver)], True)
+    coordinator = MagiQtouchCoordinator(hass, driver)
+    await driver.initial_refresh()
+    await coordinator.async_config_entry_first_refresh()
+    if driver.current_system_state.NoOfZones == 0:
+        async_add_entities([MagiQtouch(entry.entry_id, driver, coordinator, 0)], True)
+    else:
+        async_add_entities([MagiQtouch(entry.entry_id, driver, coordinator, i) for i in range(driver.current_system_state.NoOfZones)], True)
+
+
+class MagiQtouchCoordinator(DataUpdateCoordinator):
+    """ An update coordinator that handles updates for the entire MagiQtouch integration. """
+
+    controller: MagiQtouch_Driver
+
+    def __init__(self, hass, controller):
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="MagiQtouch",
+            update_interval=timedelta(seconds=10),
```

Comment:
I'm assuming this was intended to be set to `SCAN_INTERVAL` defined above.

Reply:
Whoops, you would be correct.